theano/__init__.py

```diff
@@ -195,11 +191,11 @@ def sparse_grad(var):
     ret = var.owner.op.__class__(sparse_grad=True)(*var.owner.inputs)
     return ret
 
-# This cannot be done in tensor/__init__.py due to a circular dependency -- randomstreams
-# depends on raw_random which depends on tensor.  As a work-around, we import RandomStreams
-# here and inject an instance in tensor.
-from theano.tensor.randomstreams import RandomStreams
+# This cannot be done in tensor/__init__.py due to a circular
+# dependency -- shared_randomstreams depends on raw_random which
+# depends on tensor.  As a work-around, we import RandomStreams here
+# and inject an instance in tensor.
+from theano.tensor.shared_randomstreams import RandomStreams
```

Comment:
I think we should just remove the old interface. If we make it now, we can't change de default rng later without breaking code. We can't reuse the numpy default rng on GPU, so doing that now is bad. But we could default to mrg. I think that make more sense. But we would need to build the interface to choose mrg or numpy or curand. So I prefer to let this to later. At least, not in this PR that remove deprecated stuff. This create a new interface with a new semantic.